src/main/java/com/gdsc/blog/article/service/ArticleService.java

```diff
@@ -0,0 +1,160 @@
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ArticleService {
+
+    private final UserRepository userRepository;
+    private final ArticleRepository articleRepository;
+    private final ArticleMapper articleMapper;
+
+    /**
+     * 게시글 생성
+     *
+     * @param articleCreateDto 게시글 생성 DTO
+     * @param username         작성자
+     */
+    public ArticleDto createArticle(
+            ArticleCreateDto articleCreateDto,
+            String username) {
+
+        // 현재 시간
+        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+
+        // 유저 정보
+        User user = userRepository.findByUsername(username).orElseThrow();
+
+        // 게시글 생성
+        Article article = Article.builder()
+                .title(articleCreateDto.getTitle())
+                .content(articleCreateDto.getContent())
+                .user(user)
+                .createDate(timestamp)
+                .modifyDate(timestamp)
+                .build();
+
+        return articleMapper.toDto(articleRepository.save(article));
+    }
+
+    /**
+     * 최근 게시글 목록 조회
+     *
+     * @return 게시글 목록
+     */
+    public List<ArticleDto> getRecentArticle() {
+        // get recent article list
+        List<Article> articles = articleRepository.findAll(Sort.by(Sort.Direction.DESC, "createDate")); // 최신순으로 정렬
+
+        // article list to article dto list
+        List<ArticleDto> articleList = new ArrayList<>();
+        articles.forEach(article -> articleList.add(articleMapper.toDto(article)));
+
+        return articleList;
+    }
+
+    /**
+     * 유저 게시글 목록 조회
+     *
+     * @return 게시글 목록
+     */
+    public List<ArticleDto> getUserArticle(User user) {
+        // get user article list
+        List<Article> articles = articleRepository.findByUser(user);
+        log.info("articles: {}", articles);
+
+        // article list to article dto list
+        List<ArticleDto> articleList = new ArrayList<>();
+        articles.forEach(article -> articleList.add(articleMapper.toDto(article)));
+
+        return articleList;
+    }
+
+    /**
+     * id로 게시글 조회
+     *
+     * @param idx 게시글 id
+     * @return Article 객체
+     */
+    public ArticleDto getArticleById(Long idx) { //get article by id
+        Optional<Article> article = articleRepository.findById(idx);
+
+        if (article.isPresent()) {
+            return articleMapper.toDto(article.get());
+        } else {
+            throw new IllegalArgumentException("Not found Article by id");
+        }
+    }
+
+    public List<ArticleDto> findArticleByTitle(String title) {
+        List<Article> articles = articleRepository.findByTitleLike(title);
+
+        List<ArticleDto> articleList = new ArrayList<>();
+        articles.forEach(article -> articleList.add(articleMapper.toDto(article)));
+
+        return articleList;
+    }
+
+    /**
+     * 게시글 수정
+     *
+     * @param id   게시글 id
+     * @param dto  게시글 수정 DTO
+     * @param user 작성자
+     * @return Article 객체
```

Comment:
각 파라미터에 대한 조금 더 자세한 설명이 들어가면 좋을 듯 합니다!
또, 던져지는 예외에 대한 설명도 작성되면 좋을 거 같습니다.

```suggestion
     * @param user 게시글 수정 요청 사용자
     * @return Article 수정된 게시글 내용을 담고 있는 객체
     * @throws IllegalArgumentException 게시글 작성자와 로그인한 유저가 같지 않을 시
     * @throws NoSuchElementException 게시글 또는 유저를 찾을 수 없을 시
```